tests/unit/jbi/test_router.py

```diff
@@ -10,31 +10,71 @@
+            mocked_bz_func.return_value = private_webhook_request_example.bug
+            with TestClient(app) as anon_client:
+                # https://fastapi.tiangolo.com/advanced/testing-events/
 
-        invalid_request_logs = caplog.records[1]
-        assert invalid_request_logs.name == "ignored-requests"
+                response = anon_client.post(
+                    "/bugzilla_webhook", data=private_webhook_request_example.json()
+                )
+                assert response
+                assert response.status_code == 200
+                assert (
+                    response.json()["error"]
+                    == "private bugs are not valid for this action"
+                )
 
-        assert invalid_request_logs.msg == "ignore-invalid-request: %s"
-        assert invalid_request_logs.args
-        for arg in invalid_request_logs.args:
-            assert isinstance(arg, IgnoreInvalidRequestError)
-            assert str(arg) == "private bugs are not valid"
+                invalid_request_logs = caplog.records[1]
+                assert invalid_request_logs.name == "ignored-requests"
+
+                assert invalid_request_logs.msg == "ignore-invalid-request: %s"
+                assert invalid_request_logs.args
+                for arg in invalid_request_logs.args:
+                    assert isinstance(arg, IgnoreInvalidRequestError)
+                    assert str(arg) == "private bugs are not valid for this action"
+
+
+def test_private_request_is_allowed(
+    caplog, webhook_request_example: BugzillaWebhookRequest
+):
+    private_webhook_request_example = webhook_request_example
+    private_webhook_request_example.bug.is_private = True  # type: ignore
+    test_action = Action.parse_obj(
+        {"action": "tests.unit.jbi.noop_action", "allow_private": True}
+    ).dict()
+
+    with mock.patch("src.jbi.router.extract_current_action") as mocked_extract_action:
+        mocked_extract_action.return_value = test_action
+        with mock.patch("src.jbi.router.getbug_as_bugzilla_object") as mocked_bz_func:
+            mocked_bz_func.return_value = private_webhook_request_example.bug
+            with mock.patch("tests.unit.jbi.noop_action.init") as mocked_action_init:
+                mocked_action = MagicMock()
+                mocked_action.return_value = dict()
+                mocked_action_init.return_value = mocked_action
+                with TestClient(app) as anon_client:
+                    # https://fastapi.tiangolo.com/advanced/testing-events/
+
+                    response = anon_client.post(
+                        "/bugzilla_webhook", data=private_webhook_request_example.json()
+                    )
+                    assert response
+                    assert response.status_code == 200
+
+                    assert mocked_action.call_count == 1
+                    assert mocked_action.call_args.kwargs["payload"].bug.id == 654321
```

Comment:
Alternatively, since the noop action returns the input parameters and payload, you can remove the mocking and read them from the JSON response

Reply:
Experimenting with this made me find a few issues with the action as-is.

Returning the payload from the action turns out to be a problem because we then attempt to JSON encode the payload and we fail with `Object of type BugzillaWebhookRequest is not JSON serializable`. I could convert it to a dict first but then we fail with `Object of type datetime is not JSON serializable`.

In the change I'm pushing I return the payload as a json string. That has the issue that we effectively have to double parse the response in order to get at it but it does work. Let me know if you'd prefer something else.

Reply:
Yes, this is fine I think! 

It's good as it is, but if you want to remove the call to `parse_raw()`  in the test and have something equivalent to `.dict()`, you could try to return `{..., "payload": json.loads(payload.json())}` from the action. I'm not sure but dates should become strings in that case.